kernel/src/engine/arrow_expression.rs

```diff
@@ -319,7 +319,8 @@ fn evaluate_expression(
                 Equal => |l, r| eq(l, r).map(wrap_comparison_result),
                 NotEqual => |l, r| neq(l, r).map(wrap_comparison_result),
                 Distinct => |l, r| distinct(l, r).map(wrap_comparison_result),
-                _ => return Err(Error::generic("Invalid expression given")),
+                // NOTE: [Not]In was already covered above
+                In | NotIn => return Err(Error::generic("Invalid expression given")),
```

Comment:
nit: This isn't really an invalid operation, it's just that we expected to have matched above, so maybe:
```suggestion
                In | NotIn => return Err(Error::internal("Encountered In/NotIn in general BinaryOperation match")),
```